### ISSUE TITLE
[Model] Ability to exclude fields from generated input types

### ISSUE THREAD

**Issue by craicoverflow** [aerogear/graphback#1023]
There should be additional annotations to exclude certain fields from being added to input types in the final schema.

```gql
""@model"""
type User {
  id: ID!
  name: String
  """
  @fields.ignore
  """
  password: String
  """
  @input.ignore
  """
  fieldIDontWantToEdit: Int
}
```

would generate the following input types:

```gql
input UserFields {
  id: ID
  name: String
  fieldIDontWantToEdit: Int
}

input UserInput {
  id: ID
  name: String
  password: String
}
```

**Comment by machi1990** [comment]
With the `@ignore` annotation, we'd ignore the field:
- From generated mutation inputs 
- From generated filter inputs
- From query selection set
- From database migration
  
Anything missing? @craicoverflow 

Also cc @wtrocki for diversity of ideas.

**Comment by craicoverflow** [comment]
> From generated mutation inputs

It should be included in the generated mutation inputs. Think of a `password` field - you want to create and update this, but do not want to necessarily be able to retrieve or filter on it.

> From database migration

It should still be created in the database - we have `@db(skip: true)` to ignore it in migrations and these can be combined to that effect.

Think of a `password` field as an example. You want it to be created in the database but do not want it to be selectable by the user at any stage.

We may need a combination of annotations - I think we should check out how other libraries do this, as it is probably quite a common use case in Entity Framework, Hibernate, JPA etc

**Comment by machi1990** [comment]
> > From generated mutation inputs
> 
> It should be included in the generated mutation inputs. Think of a `password` field - you want to create and update this, but do not want to necessarily be able to retrieve or filter on it.

Makes sense for the password field. But generally, If field is not excluded from the input, it will be part of the payload that's going to be stored in db. This will likely crash as the column might be missing from database.   
 
> > From database migration
> 
> It should still be created in the database - we have `@db(skip: true)` to ignore it in migrations and these can be combined to that effect.
> Think of a `password` field as an example. You want it to be created in the database but do not want it to be selectable by the user at any stage.

Instead of having several annotations that's does the same thing, I'd very much see us offering one annotation. Possibly we can make it take some arguments. 

> We may need a combination of annotations - I think we should check out how other libraries do this, as it is probably quite a common use case in Entity Framework, Hibernate, JPA etc

I'd be much in favour of having one annotation.

**Comment by craicoverflow** [comment]
Something like this?

```gql
type User {
  """@ignore(create: true, update: true, find: true, get: true)"""
  password: String
}
```

**Comment by machi1990** [comment]
> Something like this?
> 
> ```graphql
> type User {
>   """@ignore(create: true, update: true, find: true, get: true)"""
>   password: String
> }
> ```

```graphql
type User {
   """@ignore(mutate: true, query: true)"""
   password: String 
}
```

Maybe something like that. But we'd need to see how this will affect migration. Do we maintain this and the `@db(skip: true)`? Do we see user having to use both of these annotations? I'd say no.

**Comment by craicoverflow** [comment]
If both `mutate` and `query` are true it could equate to `@db(skip: true)`.

And just doing `@ignore` could default both to true.

**Comment by machi1990** [comment]
This fits nice with https://github.com/aerogear/graphback/pull/1730. I've a couple of thinking to do and I'll jump into this issue.
